src/view/ValaxyProvider.ts

```diff
@@ -1,12 +1,35 @@
+import { Buffer } from 'node:buffer'
 import type { ProviderResult, TreeDataProvider } from 'vscode'
-import { EventEmitter } from 'vscode'
+import { EventEmitter, Uri, window, workspace } from 'vscode'
 import { ctx } from '../ctx'
 import { PostItem } from './PostItem'
 
 export class PostsProvider implements TreeDataProvider<PostItem> {
   private _onDidChangeTreeData = new EventEmitter<PostItem | undefined>()
   readonly onDidChangeTreeData = this._onDidChangeTreeData.event
 
+  async add(): Promise<void> {
+    const filename = await window.showInputBox({
+      placeHolder: 'Type the filename of the new post',
+    })
+
+    if (filename) {
+      const uri = workspace.workspaceFolders?.[0].uri
+      if (!uri)
+        return
+      const filePath = Uri.joinPath(uri, '/posts', `${filename}.md`)
+      const now = new Date().toISOString()
+      await workspace.fs.writeFile(filePath, Buffer.from(`---
+title: 
+date: ${now}
+updated: ${now}
+tags:
+  - 
+categories: 
+---\n`))
+    }
+  }
```

Comment:
However, I'm not sure where to put these lines of code for the template of the new posts, and I feel like it would be abrupt if I just put them in directly.